### ISSUE TITLE
[import/no-duplicates] False positive on imports in different typescript ambient modules

### ISSUE THREAD

**Issue by chbdetta** [benmosher/eslint-plugin-import#2091]
In an ambient file `global.d.ts`
```
declare module 'A' {
  import { ReactNode } from 'react';
}

declare module 'B' {
  import { ReactNode } from 'react';
}
```

`import/no-duplicates` complains about importing 'react' twice.

**Comment by ljharb** [comment]
Duplicate of #2054.

**Comment by chbdetta** [comment]
Is this a duplicate of #2054?
That issue is about importing the same library at the root level module. But here the importing happens in two different typescript ambient modules although they are in the same file.

**Comment by ljharb** [comment]
I think it's the same issue in that the problem is the typescript resolver pointing multiple modules to the same d.ts file, masking the actual value-space location that should be checked.

**Comment by chbdetta** [comment]
However, the plugin here seems to treat the `import` in two different typescript modules (`A` and `B`) as in one module.

I am expecting the plugin to treat the scope for module `A` and module `B` as they are defined in different files.

**Comment by ljharb** [comment]
Right - that's the issue in #2054. Both modules resolved to the same file, so the plugin considers them the same module.
